lib/rubocop/cop/rspec/identical_equality_assertion.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for equality assertions with identical expressions on both sides.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo.bar).to eq(foo.bar)
+      #   expect(foo.bar).to eql(foo.bar)
+      #
+      #   # good
+      #   expect(foo.bar).to eq(2)
+      #   expect(foo.bar).to eql(2)
+      #
+      class IdenticalEqualityAssertion < Base
+        MSG = 'Identical equality can never fail.'
```

Comment:
This is not correct.
`randomizer.next_rand == randomizer.next_rand` has a chance to fail.
And `factorial.next == factorial.next` will only pass for the first two calls.

Do you mind to change the wording?

Reply:
How about:

`Identical expressions on both sides of the equality may indicate a flawed test`.

Reply:
💯